enterprise/views.py

```diff
@@ -806,22 +821,22 @@ def post(self, request, enterprise_uuid, course_id):
                 break
 
         if not selected_course_mode:
-            return self.get_enterprise_course_enrollment_page(request, enterprise_customer, course, course_modes)
+            return self.get_enterprise_course_enrollment_page(request, enterprise_customer, course, course_modes,
+                                                              enterprise_course_enrollment)
 
         user_consent_needed = is_consent_required_for_user(enterprise_customer_user, course_id)
         if not selected_course_mode.get('premium') and not user_consent_needed:
             # For the audit course modes (audit, honor), where DSC is not
             # required, enroll the learner directly through enrollment API
             # client and redirect the learner to LMS courseware page.
-            with transaction.atomic():
```

Comment:
I believe previously we wanted to revert the db transaction for creating `EnterpriseCourseEnrollment` object in case the external HTTP call to `EnrollmentApiClient` fails. Since enterprise is being installed as a django app in LMS so this call isn't external but we can decide if we need to avoid committing the `EnterpriseCourseEnrollment` in db just in case the API call fails.  @haikuginger can provide some insight.